cortex/feature_types.py

```diff
@@ -75,24 +76,32 @@ def _wrapper2(*args, **kwargs):
                 for file in [f for f in os.listdir(cache_dir) if f[-7:] == '.cortex']:  # .lamp
                     path = cache_dir + '/' + file
                     saved = dict(zip(['name', 'id', 'start', 'end'], file.split('.')[0].split('_')))
-                    saved['start']=int(saved['start'])
-                    saved['end']=int(saved['end'])
+                    saved['start'] = int(saved['start'])
+                    saved['end'] = int(saved['end'])
                     if name.split('.')[-1] == saved['name']:
                         if saved['start'] <= kwargs['start'] and saved['end'] >= kwargs['end']:
                             _result = pickle.load(open(path, 'rb'))
                             found = True
                             log.info('Using saved raw data...')
                             break
+                            
                 if not found:
                     log.info('No saved raw data found, getting new...')
                     _result = func(*args, **kwargs)
-
                     pickle_path = (cache_dir + '/' +
                                    name.split('.')[-1] + '_' + 
                                    kwargs['id'] + '_' +
                                    str(kwargs['start']) + '_' +
                                    str(kwargs['end']) + '.cortex')
-                    pickle.dump(_result, open(pickle_path, 'wb'))
+                    
+                    if kwargs.get('compression') is not None:
+                        pickle_path += '.' + kwargs.get('compression')
```

Comment:
Looks like this is being confused for a feature function parameter?

Reply:
The reason is that for the uncompressed pickles we are not saving the extension (`.cortex` instead of `.cortex.pkl`) but for compressed versions we are. (We can change that though)

Reply:
Ah I see the logic here. Thoughts:
1. `kwargs` should be used only for feature function parameters, and since `compression` doesn't apply to `gps` or `screen_state`, etc. it may need a new home? (Env vars are one possible solution but not the definitive one!)
2. I like how elegant this approach is! If you specify `bz2` or `gz`, the compressor will know what to do AND it's going to definitely be the right filename extension... but maybe we need a sanity check/assertion to confirm that `compression in ['bz2', 'gz', ...]`, otherwise it could lead to undefined results/crash in the Cortex handler code?

Reply:
1.  The use of `kwargs` is to allow the user to change a variable without changing an env var (which may affect other sessions in the IDE). I agree that it's best to only use it for the feature function parameter (and not as a parameter in the decorator). So yes -- maybe it is a fair expectation for users to manage their env vars appropriately?

2. I agree -- I will set an assertion to check the assertion type

Reply:
> allow the user to change a variable without changing an env var (which may affect other sessions in the IDE)
Thanks for this important point -- there's no good in between right now that I can think of... let's use the `kwargs` approach for now with a quick `FIXME`/`TODO` comment for later. .

> So yes -- maybe it is a fair expectation for users to manage their env vars appropriately?

I think this is fair to say! As we discuss docs this week let's be sure to add a disclaimer about managing environment correctly (or rather, "don't micromanage, configure the bare minimum, and just let cortex do the rest").